### ISSUE TITLE
Remove the remaining `any` checks from a result

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#446]
When a rule such as labels has both `any` and `none` checks, if the `any` has a pass, but the rule fails because of the `none`, all check results from `any` must be removed, not just the one check that passed. We saw this problem in the following code:

```html
<input type="text" id="fff">
<label for="fff">a</label>
<label for="fff">b</label>
```
